### ISSUE TITLE
luasnmp not work with net-snmp 5.8, ubuntu  20.04

### ISSUE THREAD

**Issue by mendisobal** [hleuwer/luasnmp#1]
in exampes file walk.lua says: " unable to open session with netsnmp:"

**Comment by hleuwer** [comment]
My last release was for net-snmp 5.7. Did you compile and link against net-snmp 5.8 and did this work?

**Comment by mendisobal** [comment]
I am using net-snmp 5.8 included with ubuntu 20.04
deb package net-snmp.
luasnmp compiled but didn't work.
I would like to know what is needed for luasnmp to work with net-snmp 5.8

**Comment by bdolez** [comment]
Hi,
I have the same issue with samples on open() calls. The only error is "netsnmp:".

```
local snmp = require "snmp"
hub1, err = snmp.open{peer = "127.0.0.1", community = "public", version = SNMPv2c}
assert(hub1, err)
```

I'm using LUA 5.3, Net-SNMP 5.8 on Fedora.

I had the same issue with previous version 1.0.6 of luasnmp.

```
ben@loki:luasnmp$ make clean
cd src && make clean
make[1]: Entering directory '/home/ben/devel/lua/luasnmp/src'
rm -f nm_snmp.o nm_util.o nm_trap.o nm_mib.o nm_c64.o except.o snmp.so.1.0.7 
rm -rf ../snmp
rm -rf makefile.deps
make[1]: Leaving directory '/home/ben/devel/lua/luasnmp/src'
ben@loki:luasnmp$ make clean
cd src && make clean
make[1]: Entering directory '/home/ben/devel/lua/luasnmp/src'
rm -f makefile.deps
test -z "nm_snmp.o nm_util.o nm_trap.o nm_mib.o nm_c64.o except.o" || gcc -MM -DUSE_SNMPTRAPD  -DVERSION='"1.0.7"' -DSYSTEM='"Linux"' -DLinux -I. -I/usr/include -I/usr/include/lua5.3 -fPIC -Wall -O2 nm_snmp.c nm_util.c nm_trap.c nm_mib.c nm_c64.c except.c >> makefile.deps
rm -f nm_snmp.o nm_util.o nm_trap.o nm_mib.o nm_c64.o except.o snmp.so.1.0.7 
rm -rf ../snmp
rm -rf makefile.deps
make[1]: Leaving directory '/home/ben/devel/lua/luasnmp/src'

```

```
ben@loki:lua$ SNMPCONFPATH="." MIBDIR='.' MIBS='.' strace -e file lua snmp-bulk-walk.lua
execve("/usr/bin/lua", ["lua", "snmp-bulk-walk.lua"], 0x7fff1dce8608 /* 87 vars */) = 0
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/liblua-5.3.so", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libreadline.so.8", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libtinfo.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "snmp-bulk-walk.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/snmp.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/lua/5.3/snmp/init.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/snmp.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/snmp/init.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp.lua", O_RDONLY) = 3
openat(AT_FDCWD, "./snmp.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/snmp/core.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/lua/5.3/snmp/core/init.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/snmp/core.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/snmp/core/init.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp/core.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp/core/init.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/snmp/core.so", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/lua/5.3/loadall.so", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp/core.so", O_RDONLY) = 3
openat(AT_FDCWD, "./snmp/core.so", O_RDONLY|O_CLOEXEC) = 3
getcwd("/home/ben/devel/lua", 128)      = 20
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libnetsnmp.so.35", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libssl.so.1.1", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libcrypto.so.1.1", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
access("/etc/system-fips", F_OK)        = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/lua/5.3/pl.lua", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/init.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/init.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/import_into.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/import_into.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/utils.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/utils.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/compat.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/compat.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/pretty.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/pretty.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/lexer.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/lexer.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/stringx.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/usr/share/lua/5.3/pl/stringx.lua", O_RDONLY) = 3
openat(AT_FDCWD, "/var/lib/net-snmp/mib_indexes", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 3
openat(AT_FDCWD, "/var/lib/net-snmp/mib_indexes/0", O_RDONLY) = 4
openat(AT_FDCWD, "/home/ben/.snmp/mibs", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
stat("/var/lib/net-snmp/mib_indexes/0", {st_mode=S_IFREG|0644, st_size=2443, ...}) = 0
stat("/usr/share/snmp/mibs", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
openat(AT_FDCWD, "/var/lib/net-snmp/mib_indexes/0", O_RDONLY) = 3
MIB search path: /home/ben/.snmp/mibs:/usr/share/snmp/mibs
openat(AT_FDCWD, "./snmp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmpapp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
stat(".", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
openat(AT_FDCWD, "./snmp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
stat(".", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
openat(AT_FDCWD, "./snmpapp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmpapp.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/var/lib/net-snmp/cert_indexes", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 3
stat("./tls/ca-certs", 0x7fffa05ffe80)  = -1 ENOENT (No such file or directory)
stat("./tls/certs", 0x7fffa05ffe80)     = -1 ENOENT (No such file or directory)
stat("./tls/private", 0x7fffa05ffe80)   = -1 ENOENT (No such file or directory)
stat(".", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
openat(AT_FDCWD, "./snmp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmp.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
stat(".", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
openat(AT_FDCWD, "./snmpapp.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./snmpapp.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
stat(".", {st_mode=S_IFDIR|0775, st_size=4096, ...}) = 0
openat(AT_FDCWD, "./hosts/127.0.0.1.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "./hosts/127.0.0.1.local.conf", O_RDONLY) = -1 ENOENT (No such file or directory)
lua: snmp-bulk-walk.lua:13: netsnmp: 
stack traceback:
	[C]: in function 'assert'
	snmp-bulk-walk.lua:13: in main chunk
	[C]: in ?
+++ exited with 1 +++

```

**Comment by hleuwer** [comment]
Fixed and upgraded to version 1.0.8 using net-snmp 5.9. Tested only on MacOs for now.
